src/glyphs.cpp

```diff
@@ -169,60 +169,81 @@ NAN_METHOD(Range) {
-        if (face_error) {
-            baton->error_name = std::string("could not open font file");
+        if (face_) FT_Done_Face(face_);
+    }
+
+    FT_Face face_;
+};
+
+void LoadAsync(uv_work_t* req) {
+    LoadBaton* baton = static_cast<LoadBaton*>(req->data);
+    try {
+        FT_Library library = nullptr;
+        ft_library_guard library_guard(library);
+        FT_Error error = FT_Init_FreeType(&library);
+        if (error) {
+            /* LCOV_EXCL_START */
+            baton->error_name = std::string("could not open FreeType library");
             return;
+            /* LCOV_EXCL_END */
         }
-        std::set<int> points;
-        if (num_faces == 0)
-            num_faces = ft_face->num_faces;
-        FT_ULong charcode;
-        FT_UInt gindex;
-        charcode = FT_Get_First_Char(ft_face, &gindex);
-        while (gindex != 0) {
-            charcode = FT_Get_Next_Char(ft_face, charcode, &gindex);
-            if (charcode != 0) points.emplace(charcode);
-        }
+        FT_Face ft_face = 0;
+        int num_faces = 0;
+        for ( int i = 0; ft_face == 0 || i < num_faces; ++i )
+        {
+            ft_face_guard face_guard(ft_face);
+            FT_Error face_error = FT_New_Memory_Face(library, reinterpret_cast<FT_Byte const*>(baton->font_data), static_cast<FT_Long>(baton->font_size), i, &ft_face);
+            if (face_error) {
+                baton->error_name = std::string("could not open font file");
+                return;
+            }
+            if (num_faces == 0) {
+                num_faces = ft_face->num_faces;
+            }
 
-        std::vector<int> points_vec(points.begin(), points.end());
+            if (ft_face->family_name || ft_face->style_name) {
+                std::set<int> points;
+                FT_ULong charcode;
+                FT_UInt gindex;
+                charcode = FT_Get_First_Char(ft_face, &gindex);
+                while (gindex != 0) {
+                    charcode = FT_Get_Next_Char(ft_face, charcode, &gindex);
+                    if (charcode != 0) points.emplace(charcode);
+                }
 
-        if (ft_face->style_name) {
-            baton->faces.emplace_back(ft_face->family_name, ft_face->style_name, std::move(points_vec));
-        } else {
-            baton->faces.emplace_back(ft_face->family_name, std::move(points_vec));
-        }
+                std::vector<int> points_vec(points.begin(), points.end());
 
-        if (ft_face) {
-            FT_Done_Face(ft_face);
+                if (ft_face->family_name && ft_face->style_name) {
+                    baton->faces.emplace_back(ft_face->family_name, ft_face->style_name, std::move(points_vec));
+                } else if (ft_face->family_name) {
+                    baton->faces.emplace_back(ft_face->family_name, std::move(points_vec));
+                } else {
+
```

Comment:
agree, fixed in https://github.com/mapbox/node-fontnik/commit/90ab6e8acf6395454205ab5e348e6e23f1d70c29